tests/test_ctranslate2.py

```diff
@@ -0,0 +1,93 @@
+Els seus tentacles ja no abastaven només a Sicília, sinó gairebé a tota l'estructura econòmica d'Itàlia, i d'usar escopetes de canons retallats, va passar a disposar d'armament més expeditiu: revòlvers del calibre .357 Magnum, fusells llança-granades, bazookas, i explosius.
+La màfia i altres societats secretes del crim organitzat van formar un sistema de vasos comunicants.
+En la lògia maçònica P-2, representada pel gran maestre Lici Gelli, hi havia ministres, parlamentaris, generals, jutges, policies, banquers, aristòcrates i fins i tot mafiosos.
+En 1992, la màfia siciliana va assassinar al jutge italià Giovanni Falcone fent esclatar mil quilograms d'explosius col·locats sota l'autopista que uneix Palerm amb l'aeroport ara anomenat Giovanni Falcone.
+Van morir ell, la seva esposa Francesca Morvilio i tres escortes.
+En 1993, cinc ex-presidents de Govern, moltíssims ministres i més de 3000 polítics i empresaris van ser acusats, processaments o condemnats per corrupció i associació amb la màfia.
+Es tractava d'un missatge de la màfia al vell Andreotti, ex-president del Govern, per no aturar l'enpresonament masiu dels seus membres.
+La màfia no perdona mai, com ja no podran testificar els banquers Michele Sindona i Roberto Calvi, dos mags de les finances del Vaticà, la màfia i altres institucions d'Itàlia.
+Van ser assassinats per un rampell de cobdícia, ja que van voler apropiar-se dels diners de la màfia.
+El capo di tutti capi és el major rang que pot haver-hi en la Cosa Nostra.
+Es tracta del cap d'una família que, en ser més poderós o per haver assassinat als altres caps de les altres famílies, s'ha convertit en el més poderós membre de la màfia.
+Un exemple d'això va ser Salvatore Maranzano , qui va ser traït per Lucky Luciano, qui finalment li va cedir el lloc ―en ser extradit per problemes amb la justícia nord-americana― a la seva mà dreta i conseller, Frank Costello.
+El don és el cap d'una família.
+"""
+
+
+@pytest.fixture
+def data_dir():
+    data_dir = DataDir("test_translate")
+    shutil.copyfile("tests/data/vocab.spm", data_dir.join("vocab.spm"))
+    return data_dir
+
+
+def download_and_cache(data_dir: DataDir, url: str, cached_filename: str, data_dir_name: str):
+    """
+    Download remote language model resources and cache them in the data directory.
+    """
+    src_dir = Path(__file__).parent.parent
+    cached_file = src_dir / "data/tests" / cached_filename
+    cached_file.parent.mkdir(parents=True, exist_ok=True)
+    if not cached_file.exists():
+        stream_download_to_file(url, cached_file)
+    shutil.copy(cached_file, data_dir.join(data_dir_name))
+
+
+def test_ctranslate2():
+    data_dir = DataDir("test_ctranslate2")
+    data_dir.mkdir("model1")
+    data_dir.create_zst("file.1.zst", text)
+
+    # Download the teacher models.
+    download_and_cache(
+        data_dir,
+        "https://storage.googleapis.com/releng-translations-dev/models/ca-en/dev/teacher-finetuned1/final.model.npz.best-chrf.npz",
```

Comment:
The translation itself might also be slow

Reply:
It takes 40 seconds to run locally. I tried training a tiny teacher, but the problem is the assertions are then junk. I'd like to at least have sensible assertions to have confidence it's working. I'm going to file a follow-up to train a real smaller teacher, since CI isn't really set up to do good results.

<img width="745" alt="image" src="https://github.com/user-attachments/assets/368aadb0-fbe9-48f0-9536-2490be384267" />

Reply:
#970